frontend/src/models/models.js

```diff
@@ -42,16 +42,22 @@ module.exports = app => app.component('models', {
   },
   methods: {
     async search() {
-      if (this.searchText) {
+      if (this.searchText && Object.keys(this.searchText).length) {
         this.filter = eval(`(${this.searchText})`);
         this.filter = EJSON.stringify(this.filter);
+        this.$router.push({ path: this.$route.path, query: { search: this.searchText }})
       } else {
         this.filter = {};
+        this.$router.push({ path: this.$route.path });
       }
-      
       await this.getDocuments();
     },
     async getDocuments() {
+      if (this.$route.query?.search) {
```

Comment:
This should just be in the `mounted` hook. Otherwise this logic will run _every_ time you load more documents, which is unnecessary